### ISSUE TITLE
shim plugin does not run multiple inputs in the plugin’s config - it only runs the first

### ISSUE THREAD

**Issue by ashishkaransingh** [influxdata/telegraf#9325]
### Relevant telegraf.conf:

```toml
custom_plugin.conf (https://github.com/influxdata/telegraf/tree/master/plugins/common/shim)
[[inputs.custom_mssql_1]]
  servers = ["Server = 10.111.00.000;Port=1433;Database=Database_1;app name=telegraf;log=1;"]
  queries = ["sql_query_from_plugin_1"]

[[inputs.custom_mssql_1]]
  servers = ["Server = 10.111.00.000;Port=1433;Database=Database_2;app name=telegraf;log=1;"]
  queries = ["sql_query_from_plugin_2"]

[[inputs.custom_mssql_1]]
  servers = ["Server = 10.111.00.000;Port=1433;Database=Database_3;app name=telegraf;log=1;"]
  queries = ["sql_query_from_plugin_3"]

Telegraf.exe.conf

[[inputs.execd]]
  command = ["C:\\Users\\*****\\go\\src\\sql-server\\custom_mssql_.exe", "--config", "C:\\Users\\*****\\go\\src\\sql-server\\custom_mssql.conf"]
  signal = "STDIN"
```

### System info:

```
Telegraf  v1.18.2
go-mssqldb v0.10.0
go 1.15
```

### Steps to reproduce:

1. running locally with telegraf.exe --config telegraf.conf --test
2. multiple inputs in the plugin’s config (custom_plugin.conf) but it only runs the first i.e ```["sql_query_from_plugin_1"]```

### Expected behavior:

All the inputs containing sql queries  under custom_plugin.conf should run
```
["sql_query_from_plugin_1"], ["sql_query_from_plugin_2"], ["sql_query_from_plugin_3"]
```
### Actual behavior:

Only the following runs, rest two are skipped
```[[inputs.custom_mssql_1]]
  servers = ["Server = 10.111.00.000;Port=1433;Database=Database_1;app name=telegraf;log=1;"]
  queries = ["sql_query_from_plugin_1"]
```
### Additional info:
- Each input is for a separate sql database
- It is very important for us because there are multiple DBs on a server for different products and we should be able to run these queries for different databases.
- It works fine if I compile the plugin in Telegraf itself, but not as a separate app

- Is this because main agent has for loop, and that the shim doesn’t 
Example:
https://github.com/influxdata/telegraf/blob/2e7b232073650883b016b7bf2ab585a75862f6b2/agent/agent.go#L188
                for _, input := range a.Config.Inputs {

**Comment by cruscio** [comment]
The goal here is to have a single plugin in the binary, And have multiple instances of the same input plugin, without having to spawn multiple processes. 

If this isn't easily implemented, at the very least the [example code comment](https://github.com/influxdata/telegraf/blob/1cfe8da93f87c7b14a91088da5187ff1fbd9e165/plugins/common/shim/example/cmd/main.go#L55) should probably be corrected to remove the indication of plurality
```
  // run the input plugin(s) until stdin closes or we receive a termination signal
  if err := shim.Run(*pollInterval); err != nil {
    fmt.Fprintf(os.Stderr, "Err: %s\n", err)
    os.Exit(1)
  }
```

**Comment by ssoroka** [comment]
Good question. the plurality was indeed left vague, and I'm not surprised it doesn't work. I think there's some missing work to multiplex the stdin/stdout pipes across all the input plugins. At the moment, while the api allows it, it only expects a single plugin per shim/executable.

**Comment by ashishkaransingh** [comment]
Thanks for your response @ssoroka!
What amount of a work does it look like to fix the bug?

**Comment by ssoroka** [comment]
I'm not even sure it's a good idea for it to work this way; I'll have to investigate first and get back to you.

**Comment by ssoroka** [comment]
Ok, I think it's technically possible, but that it would add a lot of complication to the shim code. Right now the shim only wraps a single plugin, and with this change it'd need to act more like a mini-Telegraf agent for all the plugins it manages. This doesn't make a ton of sense since it can never do different plugin types (inputs/processors/outputs), as they all communicate differently over stdin/stdout. There's also the issue that the config doesn't map directly from one execd plugin in the toml to multiple running plugins; this doesn't cause problems if you manage your own config, but it at least adds confusion. 

TL;DR: I don't think it's very likely that we would add support for running multiple plugins in a single execd app; it complicates the code and test cases significantly. My recommendation would be to run them as separate execd applications. I suspect you'll get better performance from this setup anyway.
